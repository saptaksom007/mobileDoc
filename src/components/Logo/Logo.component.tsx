import React, { PureComponent } from 'react'
import { View, Image } from 'react-native'
import styles from './Logo.styles'

interface Props {
  height: number
}

class Logo extends PureComponent<Props> {
  render() {
    const { height } = this.props
    return (
      <View style={styles.container}>
        <Image
          style={[
            styles.logoImg,
            {
              height: height / 2,
            },
          ]}
          source={require('../../assets/images/logo_docdok.png')}
        />
      </View>
    )
  }
}

export default Logo
