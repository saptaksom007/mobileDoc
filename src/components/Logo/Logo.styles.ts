import { StyleSheet } from 'react-native'
import { Layout } from 'constants/Layout'

export default StyleSheet.create({
  container: {
    // width: 100,
  },
  logoImg: {
    // flex: 1,
    // width: undefined,
    // height: undefined,
    resizeMode: 'contain',
    width: Layout.window.width * 0.2,
  },
})
