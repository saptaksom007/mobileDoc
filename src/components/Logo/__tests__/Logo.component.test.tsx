import 'react-native'
import React from 'react'
import renderer from 'react-test-renderer'
import Logo from 'components/Logo/Logo.component'

it('<Logo/> render correctly', () => {
  const tree = renderer.create(<Logo height={123} />).toJSON()
  expect(tree).toMatchSnapshot()
})
