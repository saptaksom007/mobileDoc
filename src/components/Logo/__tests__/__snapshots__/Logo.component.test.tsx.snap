// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Logo/> render correctly 1`] = `
<View
  style={Object {}}
>
  <Image
    source={1}
    style={
      Array [
        Object {
          "resizeMode": "contain",
          "width": 150,
        },
        Object {
          "height": 61.5,
        },
      ]
    }
  />
</View>
`;
